### PR TITLE
Add short taker UI 2

### LINKED ISSUE #1422: Improve the Taker UI to allow going short
Initial version of going short in the taker's UI. 

 - tiles should show the Position (Long / Short)
 - at the very least we need 2 different prices (perhaps price on the button again?)

open questions hopefully solved by this task:
  - does it make sense to be able to go long and short at the same time?
  - is the UI still understandable with the new additions or we need some bigger redesign?

note: this task is not blocked by the backend, could initially show a dummy price in the UI.

### PR DESCRIPTION
**Pull request by rishflab**
Finalised/cleaned up version of #1486

### CODE REVIEW

File: taker-frontend/src/components/ConfirmOrderModal.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    confirmRef: any;
+    quantity: number;
+    minQuantity: number;
+    askPriceAsNumber?: number;
+    margin: number;
+    leverage: number;
+    liquidationPriceAsNumber: number | undefined;
+    feeForFirstSettlementInterval: number;
+    fundingRateHourly: string;
+    fundingRateAnnualized: string;
```
Comment (thomaseizinger) [inline taker-frontend/src/components/ConfirmOrderModal.tsx]: This seems a bit excessive, do we really need all of these?

`setQuantity` for example is only used to set the input field back to `minQuantity`. This component doesn't even know about the input field so I'd set the call to `setQuantity` should happen in the callback that the parent passes in for `onClose`.

Comment (thomaseizinger) [reply taker-frontend/src/components/ConfirmOrderModal.tsx]: We should do a similar analysis for all other props.

File: taker-frontend/src/types.ts [hunk 2]
```diff
@@ -235,6 +235,7 @@ export enum StateGroupKey {
 export interface CfdOrderRequestPayload {
     order_id: string;
     quantity: number;
+    position: string;
```
Comment (thomaseizinger) [inline taker-frontend/src/types.ts:238]: I think I'd have a slight preference for a 2nd endpoint instead of adding a field here.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks! Nice commit structure. I got no opinion on the colour change but I think the `ConfirmOrderModal` component has too many props and we should be able to simplify some of those!

**Comment by da-kami** [comment]
I am a bit confused, is it supposed to look like this?

image

Several points  addressed in https://github.com/itchysats/itchysats/issues/1422 were not actioned. 
I don't think this closes this ticket. 

- There have to be two prices (one for short and one for long). I only see one price (the long price) - am I missing something?
- The blinking price should likely go away. The idea was to simplify the two prices and put them on the button for now. We can experiment with two blinking prices, but am not sure about that.
- The required margin for short and long is different (because the prices are different), so we have to show short and long margin

The position colors can be improved, but that is not super important and can be done later. I was thinking about changing the color palette a bit more.

**Comment by rishflab** [comment]
> I am a bit confused, is it supposed to look like this?
> 
> image
> 
> Several points addressed in #1422 were not actioned. I don't think this closes this ticket.
> 
>     * There have to be two prices (one for short and one for long). I only see one price (the long price) - am I missing something?
> 
>     * The blinking price should likely go away. The idea was to simplify the two prices and put them on the button for now. We can experiment with two blinking prices, but am not sure about that.
> 
>     * The required margin for short and long is different (because the prices are different), so we have to show short and long margin
> 
> 
> The position colors can be improved, but that is not super important and can be done later. I was thinking about changing the color palette a bit more.

I think you are right, removed the closes PR tag. I can add a two prices in this PR (use a dummy price for the short). Did not understand how the price worked (backend changes were not complete and I was looking at the backend for guidance).

**Comment by thomaseizinger** [comment]
Even though it does not complete the ticket, I think we can still merge these commits independently I guess?

What do you think @da-kami ?

File: taker-frontend/src/types.ts (hunk 2 above)
Comment (klochowicz) [reply taker-frontend/src/types.ts:238]: I don't think we need 2 endpoints at all - the `order_id` unambiguously denotes the order. different positions will need different order_ids.

Comment (rishflab) [reply taker-frontend/src/types.ts:238]: yep I don't see the need, the position is in the payload.

**Review by thomaseizinger — APPROVED** [review]
concept ACK

The refactoring can be squashed into the original commit that creates the modal.

**Review by klochowicz — APPROVED** [review]
cool, let's squash this and let's get this in! thanks :)

**Comment by thomaseizinger** [comment]
@rishflab Some typescript errors are failing the build.

**Pull request merged**